.github/workflows/clp-dependency-image-build.yaml

```
@@ -26,9 +26,23 @@ jobs:
 
       - name: Workaround actions/runner-images/issues/6775
         run: chown $(id -u):$(id -g) -R .
+
+      - name: Filter relevant changes
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # Consider changes between the previous commit and this one
+          base: ${{ github.ref_name }}
+          filters: |
+            src:
+              - '.github/workflows/clp-dependency-image-build.yaml'
+              - 'components/core/tools/docker-images/clp-env-base-focal/**'
+              - 'components/core/tools/scripts/lib_install/*.sh'
+              - 'components/core/tools/scripts/lib_install/ubuntu-focal/**'
     
       - name: Build and Push Ubuntu Focal Docker Image
         uses: ./.github/actions/clp-docker-build-push-action
+        if: steps.filter.outputs.src == 'true'
```

```suggestion
        if: 'true' == steps.filter.outputs.src
```
Nit. not sure if we can do this in `yaml`. Same below

---

Unfortunately, GitHub doesn't allow that.